src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,124 @@
+        numTask--;
+        System.out.println("     Noted. I've removed this task:\n"
+                + "     " + removed.printDesc() + "\n"
+                + "     Now you have " + numTask +" tasks in the list.");
+    }
+    public static void listTask() {
+        System.out.println("     Here are the tasks in your list:\n");
+        for (int a = 0; a < numTask; a++) {
+            System.out.println("     " + (a + 1) + ". " + taskArray.get(a).printDesc());
+        }
+    }
+    public static void todoTask(String i) throws DukeArgumentException {
+        String[] taskDetails = parser.commandSplit(i);
+        if (taskDetails.length != 2) {
+            throw new DukeArgumentException("     OOPS!!! The description of todo cannot be empty :(");
+        }
+        taskArray.add(new Todo(taskDetails[1]));
+        taskArray.get(numTask).printMessage(numTask);
+        numTask++;
+    }
+    public static void deadlineTask(String i) throws DukeArgumentException {
+        String[] taskDetails = parser.commandSplit(i);
+        if (taskDetails.length != 2) {
+            throw new DukeArgumentException("     OOPS!!! The description of deadline cannot be empty :(");
+        }
+        String[] deadlineDetails = parser.deadlineDetails(taskDetails[1]);
+        if (deadlineDetails.length != 2) {
+            throw new DukeArgumentException("     OOPS!!! Where is the deadline time?");
+        }
+        taskArray.add(new Deadline(deadlineDetails[0], deadlineDetails[1]));
+        taskArray.get(numTask).printMessage(numTask);
+        numTask++;
+    }
+    public static void eventTask(String i) throws DukeArgumentException {
+        String[] taskDetails = parser.commandSplit(i);
+        if (taskDetails.length != 2) {
+            throw new DukeArgumentException("     OOPS!!! The description of event cannot be empty :(");
+        }
+        String[] eventDetails = parser.eventDetails(taskDetails[1]);
+        if (eventDetails.length != 3) {
+            throw new DukeArgumentException("     OOPS!!! The details for the event is missing!");
+        }
+        taskArray.add(new Event(eventDetails[0], eventDetails[1].substring(5),
+                eventDetails[2].substring(3)));
+        taskArray.get(numTask).printMessage(numTask);
+        numTask++;
+    }
+    protected ArrayList<Task> getTaskArray() {
+        return taskArray;
+    }
+    protected void findTask(String info) {
+        String keyword = (parser.commandSplit(info))[1];
+        ArrayList<Task> keywordTasks = new ArrayList<>();
+        for (int i = 0; i < numTask; i++) {
+            if (taskArray.get(i).getDesc().contains(keyword)) {
+                keywordTasks.add(taskArray.get(i));
+            }
+        }
+        System.out.println("     Here are the matching tasks in your list:\n");
+        for (int a = 0; a < keywordTasks.size(); a++) {
+            System.out.println("     " + (a + 1) + ". " + keywordTasks.get(a).printDesc());
```

Comment:
I would think for better abstraction it would be better to place the print statements in the Ui class